tests/foreman/ui_airgun/test_hostcollection.py

```diff
@@ -14,11 +14,162 @@
 
 :Upstream: No
 """
+import time
+
 from nailgun import entities
 
 from robottelo.api.utils import promote
-from robottelo.datafactory import gen_string
-from robottelo.decorators import tier3, upgrade
+from robottelo.config import settings
+from robottelo.constants import (
+    DISTRO_DEFAULT,
+    FAKE_0_CUSTOM_PACKAGE,
+    FAKE_0_CUSTOM_PACKAGE_NAME,
+    FAKE_0_CUSTOM_PACKAGE_GROUP,
+    FAKE_0_CUSTOM_PACKAGE_GROUP_NAME,
+    FAKE_1_CUSTOM_PACKAGE,
+    FAKE_1_CUSTOM_PACKAGE_NAME,
+    FAKE_1_YUM_REPO,
+    FAKE_2_CUSTOM_PACKAGE,
+    FAKE_2_ERRATA_ID,
+    FAKE_6_YUM_REPO,
+)
+from robottelo.datafactory import gen_string, gen_integer
+from robottelo.decorators import fixture, tier3, upgrade
+from robottelo.products import (
+    YumRepository,
+    RepositoryCollection,
+    SatelliteToolsRepository,
+)
+from robottelo.vm import VirtualMachine
+
+
+@fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
+
+
+@fixture(scope='module')
+def module_lce(module_org):
+    return entities.LifecycleEnvironment(organization=module_org).create()
+
+
+@fixture(scope='module')
+def module_repos_collection(module_org, module_lce):
+    repos_collection = RepositoryCollection(
+        distro=DISTRO_DEFAULT,
+        repositories=[
+            SatelliteToolsRepository(),
+            YumRepository(url=FAKE_1_YUM_REPO),
+            YumRepository(url=FAKE_6_YUM_REPO),
+        ]
+    )
+    repos_collection.setup_content(
+        module_org.id, module_lce.id, upload_manifest=True)
+    return repos_collection
+
+
+@fixture
+def vm_content_hosts(module_repos_collection):
+    clients = []
+    for _ in range(2):
+        client = VirtualMachine(distro=module_repos_collection.distro)
+        clients.append(client)
+        client.create()
+        module_repos_collection.setup_virtual_machine(client)
+    yield clients
+    for client in clients:
+        client.destroy()
+
+
+@fixture
+def vm_host_collection(module_org, vm_content_hosts):
+    host_ids = [
+        entities.Host().search(query={
+            'search': 'name={0}'.format(host.hostname)})[0].id
+        for host in vm_content_hosts
+    ]
+    host_collection = entities.HostCollection(
+        host=host_ids,
+        organization=module_org,
+    ).create()
+    return host_collection
+
+
+def _assert_package_installed(
```

Comment:
one of the things i like in pytest over unittest - you don't have to memorize all numerous assertions, python's builtin `assert` checks everything. Pls, let's not introduce extra assertions, but rather make this func `def package_installed` return bool so we can do `assert package_installed(...)` in tests.

Reply:
Btw `assert not package_installed()` sounds way better to me than `_assert_package_installed(expected_installed=False)`

Reply:
in this case we will wait for all retries and waist time

Reply:
For negative scenario? Imo in such case we should provide `retries`: `assert not package_installed(retries=0)`, that still sounds better to me.